### PR TITLE
Refactor Int21h into methods.

### PR DESCRIPTION
**Pull request by paladine**
Fix some behavior of the ordinals.

### CODE REVIEW

**Review by tuday2 — APPROVED** [review]
Passed 37 module test and let run for 1 hour (12 RTick's running).

File: MBBSEmu/DOS/Interrupts/Int21h.cs [hunk 1]
```diff
@@ -90,443 +90,475 @@ public void Handle()
...
+
+                Note: if CX is zero, no data is written, and the file is truncated or extended
+                    to the current position
+            */
+            var fileHandle = _registers.BX;
+            var numberOfBytes = _registers.CX;
+            var bufferPointer = new FarPtr(_registers.DS, _registers.DX);
+
+            var dataToWrite = _memory.GetArray(bufferPointer, numberOfBytes);
+            char[] toWrite = new char[numberOfBytes];
```
Comment (enusbaum) [inline MBBSEmu/DOS/Interrupts/Int21h.cs:413]: Var

File: MBBSEmu/DOS/Interrupts/Int21h.cs [hunk 2]
```diff
@@ -90,443 +90,475 @@ public void Handle()
...
+
+            if (_memory is ProtectedModeMemoryCore)
+            {
+                ProtectedModeMemoryCore protectedMemory = (ProtectedModeMemoryCore)_memory;
+                if (!protectedMemory.HasSegment(segmentToAdjust))
+                    protectedMemory.AddSegment(segmentToAdjust);
+
+                _registers.BX = 0xFFFF;
+                _registers.F = _registers.F.ClearFlag((ushort)EnumFlags.CF);
+            }
```
Comment (enusbaum) [inline MBBSEmu/DOS/Interrupts/Int21h.cs:474]: Return?

File: MBBSEmu/DOS/Interrupts/Int21h.cs [hunk 3]
```diff
@@ -564,7 +596,83 @@ private void GetDeviceInformation()
...
+            DS:DX -> ASCIZ file name
+            Return: CF set on error
+                AX = error code
+                CX = file attributes on get
+            */
+            var file = Encoding.ASCII.GetString(_memory.GetString(_registers.DS, _registers.DX, stripNull: true));
+            if (_registers.AL != 0)
+                throw new NotImplementedException();
+
+            FileInfo fileInfo = new FileInfo(file);
```
Comment (enusbaum) [inline MBBSEmu/DOS/Interrupts/Int21h.cs:623]: Var

File: MBBSEmu/DOS/Interrupts/Int21h.cs [hunk 4]
```diff
@@ -564,7 +596,83 @@ private void GetDeviceInformation()
...
+
+        private void CloseFile_0x3E()
+        {
+            /*
+            INT 21 - AH = 3Eh DOS 2+ - CLOSE A FILE WITH HANDLE
+            BX = file handle
+            Return: CF set on error
+                AX = error code
+            */
+            switch (_registers.BX)
```
Comment (enusbaum) [inline MBBSEmu/DOS/Interrupts/Int21h.cs:655]: Could do the enum cast here, don't think it matters performance wise. Would just be less code :P

**Review by enusbaum — CHANGES_REQUESTED** [review]
All I could get in before landing :P

File: MBBSEmu/DOS/Interrupts/Int21h.cs [hunk 5]
```diff
@@ -41,7 +41,7 @@ public class Int21h : IInterruptHandler
         ///
         ///     Buffer used to hold information on the current Disk / IO operation
         /// </summary>
-        private FarPtr DiskTransferArea;
+        private FarPtr DiskTransferArea = new FarPtr(0xF000, 0x1000);
```
Comment (enusbaum) [inline MBBSEmu/DOS/Interrupts/Int21h.cs:44]: `new()`

File: MBBSEmu/DOS/Interrupts/Int21h.cs [hunk 6]
```diff
@@ -90,443 +90,475 @@ public void Handle()
...
+                AX = error code
+                BX = maximum size possible for the block
+
+                Because MBBSEmu allocates blocks as 0xFFFF in length, we ignore this and proceed
+            */
+
+            var segmentToAdjust = _registers.ES;
+            var newSize = _registers.BX;
+
+            if (_memory is ProtectedModeMemoryCore)
```
Comment (enusbaum) [inline MBBSEmu/DOS/Interrupts/Int21h.cs:466]: you can combine the cast and type check here

`_memory is ProtectedModeMemoryCore protectedMemory`

**Review by enusbaum — CHANGES_REQUESTED** [review]
ok -- final review 😄

File: MBBSEmu/DOS/Interrupts/Int21h.cs (hunk 4 above)
Comment (paladine) [reply MBBSEmu/DOS/Interrupts/Int21h.cs:655]: Leaving in for now since next PR implements this fully

File: MBBSEmu/DOS/Interrupts/Int21h.cs (hunk 6 above)
Comment (paladine) [reply MBBSEmu/DOS/Interrupts/Int21h.cs:466]: nice, forgot about that

**Pull request merged**